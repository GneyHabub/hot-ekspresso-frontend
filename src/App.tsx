import { Route, Routes, Navigate } from 'react-router-dom';
import Layout from './components/Layout/Layout';
import { NotificationContainer } from './components/Notification/NotificationContainer';
import Login from './pages/login/Login';
import Tickets from './pages/tickets/Tickets';
import { useAuth0 } from "@auth0/auth0-react";
import { CircularProgress, Divider, Drawer, Icon, IconButton, List, ListItem, ListItemIcon, ListItemText } from '@mui/material';
import React from 'react';
import { Box } from '@mui/system';
import AirplaneTicketIcon from '@mui/icons-material/AirplaneTicket';
import HotelIcon from '@mui/icons-material/Hotel';
import ModeOfTravelIcon from '@mui/icons-material/ModeOfTravel';
import CloseIcon from '@mui/icons-material/Close';

const App: React.FC = () => {
  const {isAuthenticated, isLoading} = useAuth0();
  const [isSidebarOpen, setIsSidebarOpen] = React.useState(false);

  if(isLoading) {
    return (
    <Layout>
      <CircularProgress />
    </Layout>)
  }

  if(!isAuthenticated) {
    return(
      <Layout>
        <Routes>
          <Route path='/login'element={<Login/>}/>
          <Route path='*' element={<Navigate replace to="/login"/>}/>
        </Routes>
        <NotificationContainer/>
      </Layout>
    );
  }

  return (
    <>
      <Layout
        onSidebarOpen={() => setIsSidebarOpen(true)}
      >
        <Routes>
          <Route path='/'element={<Tickets/>}/>
          <Route path='*' element={<Navigate replace to="/"/>}/>
        </Routes>
        <NotificationContainer/>
      </Layout>
      <Drawer
        anchor={'left'}
        open={isSidebarOpen}
        onClose={() => setIsSidebarOpen(false)}
      >
        <Box
          sx={{
            width: 250
          }}
          role="presentation"
        >
          <Box
            sx={{
              display: "flex",
              justifyContent: "flex-end",
              padding: "5px"
            }}
          >
            <IconButton
              onClick={() => setIsSidebarOpen(false)}
            >
              <CloseIcon />
            </IconButton>
          </Box>
          <Divider />
          <List>
            <ListItem button>
              <ListItemIcon>
                <AirplaneTicketIcon />
              </ListItemIcon>
              <ListItemText primary="Tickets" />
            </ListItem>

            <ListItem button>
              <ListItemIcon>
                <HotelIcon />
              </ListItemIcon>
              <ListItemText primary="Hotels" />
            </ListItem>

            <ListItem button>
              <ListItemIcon>
                <ModeOfTravelIcon />
              </ListItemIcon>
              <ListItemText primary="Trip planning" />
            </ListItem>
          </List>
        </Box>
      </Drawer>
    </>
  )
};

export default App