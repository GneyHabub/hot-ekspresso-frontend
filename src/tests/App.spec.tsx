import App from '../App';
import {render} from '@testing-library/react'
import {BrowserRouter} from 'react-router-dom'

jest.mock('@auth0/auth0-react', () => ({
  useAuth0: () => ({
    isAuthenticated: true,
    isLoading: false
  })
}))


describe('App', () => {
  it('matches the snapshot', () => {
    const wrapper = render(<BrowserRouter><App/></BrowserRouter>);
    expect(wrapper.baseElement).toMatchSnapshot();
  })
})