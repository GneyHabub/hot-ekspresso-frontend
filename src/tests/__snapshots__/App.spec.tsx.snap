// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1yg6i26"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          
          <div
            class="MuiTypography-root MuiTypography-h6 MuiLink-root MuiLink-underlineNone css-1qqbuje-MuiTypography-root-MuiLink-root"
            href="/"
          >
            Hot Ekspresso Travel ✈️
          </div>
          
        </div>
      </header>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <form
          style="display: flex; justify-content: center;"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xg5b83-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                >
                  Login
                </span>
              </div>
            </div>
            <div
              class="MuiCardContent-root css-641s0v-MuiCardContent-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                for="standard-adornment-username"
              >
                Username
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1tcs2i1-MuiInputBase-root-MuiInput-root"
              >
                <input
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id="standard-adornment-username"
                  name="username"
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AccountCircleIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <label
                class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                for="standard-adornment-password"
              >
                Password
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1tcs2i1-MuiInputBase-root-MuiInput-root"
              >
                <input
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id="standard-adornment-password"
                  name="password"
                  type="password"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="VisibilityIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <button
                class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                id="submit"
                tabindex="0"
                type="submit"
              >
                Log In
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
